jdaviz/configs/imviz/helper.py

```diff
@@ -195,6 +199,193 @@ def link_data(self, **kwargs):
+
+            # regions: Convert to ROI.
+            # NOTE: Out-of-bounds ROI will succeed; this is native glue behavior.
+            if isinstance(region, (CirclePixelRegion, CircleSkyRegion,
+                                   EllipsePixelRegion, EllipseSkyRegion,
+                                   RectanglePixelRegion, RectangleSkyRegion)):
+                state = regions2roi(region, wcs=data.coords)
+
+                # TODO: Do we want user to specify viewer? Does it matter?
+                self.app.session.edit_subset_mode._mode = NewMode
+                self.default_viewer.apply_roi(state)
+                self.app.session.edit_subset_mode.edit_subset = None  # No overwrite next iteration # noqa
+
+            # Last resort: Masked Subset that is static
+            else:
+                im = None
+                if hasattr(region, 'to_pixel'):  # Sky region: Convert to pixel region
+                    if not has_wcs:
+                        bad_regions.append((region, 'Sky region provided but data has no WCS'))
+                        continue
+                    region = region.to_pixel(data.coords)
+
+                if hasattr(region, 'to_mask'):
+                    try:
+                        mask = region.to_mask(**kwargs)
+                        im = mask.to_image(data.shape)  # Can be None
+                    except Exception as e:  # pragma: no cover
+                        bad_regions.append((region, f'Failed to load: {repr(e)}'))
+                        continue
+
+                elif (isinstance(region, np.ndarray) and region.shape == data.shape
+                        and region.dtype == np.bool_):
+                    im = region
+
+                if im is None:
+                    bad_regions.append((region, 'Mask creation failed'))
+                    continue
+
+                # NOTE: Region creation info is thus lost.
+                try:
+                    subset_label = f'{msg_prefix} {msg_count}'
+                    state = MaskSubsetState(im, data.pixel_component_ids)
+                    self.app.data_collection.new_subset_group(subset_label, state)
+                    msg_count += 1
+                except Exception as e:  # pragma: no cover
+                    bad_regions.append((region, f'Failed to load: {repr(e)}'))
+                    continue
+
+            n_loaded += 1
+            if max_num_regions is not None and n_loaded >= max_num_regions:
+                break
+
+        n_reg_in = len(regions)
+        n_reg_bad = len(bad_regions)
+        if n_loaded == 0:
+            snack_color = "error"
+        elif n_reg_bad > 0:
+            snack_color = "warning"
+        else:
+            snack_color = "success"
+        self.app.hub.broadcast(SnackbarMessage(
+            f"Loaded {n_loaded}/{n_reg_in} regions, max_num_regions={max_num_regions}, "
+            f"bad={n_reg_bad}", color=snack_color, timeout=8000, sender=self.app))
+
+        return bad_regions
```

Comment:
I know both the updated docs and the docstring mention what the output represents, but I wonder if returning a blank list when none of the regions are bad is confusing for users who successfully use `load_regions()` but didn't  assign it to a variable. (We do this ourselves in the simple aperture photometry notebook.) To me, the most intuitive output (if any) would be the good regions. Not returning anything, like `load_data()`, also seems all right. 

What was your inspiration for doing it this way?

Reply:
My motivation is this: Returning the bad ones allow users to inspect which ones didn't load and find ways to fix them if needed. I implemented this originally for debugging but then thought maybe users would like this same feature too.

If you find it too confusing, maybe I can implement one more keyword to not return anything unless that keyword is set to some non-default value.

Please let me know if I can keep this as-is or you want me to change it. Thanks!

Reply:
That makes sense to me. I support your keyword argument idea as a good compromise.

Reply:
Done!